cypress/e2e/signUp.cy.js

```
@@ -1,12 +1,38 @@
-/// <reference types='cypress' />
-/// <reference types='../support' />
+/// <reference types="cypress" />
+/// <reference types="../support" />
+
+import SignUpPageObject from '../support/pages/signUp.pageObject';
+import HomePageObject from '../support/pages/home.pageObject';
+
+const signUpPage = new SignUpPageObject();
+const homePage = new HomePageObject();
 
 describe('Sign Up page', () => {
-  before(() => {
+  let user;
 
+  beforeEach(() => {
+    cy.task('db:clear');
+    cy.task('generateUser').then((generateUser) => {
+      user = generateUser;
+    });
   });
 
-  it('should ...', () => {
+  it('should allow register with non-existing data', () => {
+    signUpPage.visit();
+    signUpPage.typeUsername(user.username);
+    signUpPage.typeEmail(user.email);
+    signUpPage.typePassword(user.password);
+    signUpPage.clickSignUpButton();
+    homePage.assertHeaderContainUsername(user.username);
+  });
 
+  it.only('should not allow register with existing data', () => {
```

Using `.only` is a useful tool for debugging, but it should be removed before submitting your work. It causes Cypress to skip all other tests in this file, which can hide failures.